lib/media/media_source_engine.js

```diff
@@ -586,11 +586,11 @@ shaka.media.MediaSourceEngine = class {
       mimeType = this.transmuxers_[contentType].getOrginalMimeType();
     }
     if (shaka.util.TsParser.probe(uint8ArrayData)) {
-      const metadata = new shaka.util.TsParser().parse(uint8ArrayData)
-          .getMetadata();
+      const tsParser = new shaka.util.TsParser().parse(uint8ArrayData);
+      const timestampOffset =
+          reference.startTime - tsParser.getStartTime()[contentType] || 0;
```

Comment:
I don't understand why this is more correct.

I also don't understand if `|| 0` is an alternative for the second term or the difference expression.  Could you add parentheses to clarify for people like me who don't have complete operator precedence memorized?

Reply:
0 is erroneous, sorry.

The problem with this is how the sourcebuffer and timestampoffset work. When the first segment is added, it is always 0 until a segment is added to the buffer, then timestampoffset has the correct value.